native_client/alphabet.h

```diff
@@ -17,29 +18,29 @@ class Alphabet {
   Alphabet(const char *config_file) {
     std::ifstream in(config_file, std::ios::in);
     unsigned int label = 0;
+    space_label_ = -2;
```

Comment:
I don't see where ctc_beam_search_decoder knows about this magic number? It's supposed to not match any real label if there's no space label in the alphabet. Although now I realize that there's actually a subtle bug here, since space_label_ is unsigned, this is assigning a missing space label to UINT_MAX-1, which could break for an alphabet that's exactly UINT_MAX long :)

Reply:
Thinking about it I don't think this is a big deal as languages without a space label will always use character-based language models.

Reply:
Is it true that "languages without a space label will always use character-based language models"? For example Thai has an alphabet but words can also can be written without spaces. I think Javanese is the same way.

Reply:
The -2 constant would be a problem for a language that uses a word-based language model and has an alphabet that is 2^32-2 characters long. AFAIK, the CJK languages are the only ones that could possibly get close to that many characters, and they would all probably use character-level LMs. Unicode defines close to 100,000 CJK ideograms, so even then we still have a lot of margin. I can add a separate "has_space" flag to remove the in-band signaling here, but I don't think it would be a problem anytime soon.